src/rpc/blockchain.cpp

```diff
@@ -1654,26 +1654,19 @@ bool FillScRecordFromInfo(const uint256& scId, const CSidechain& info, CSidechai
         sc.pushKV("immatureAmounts", ia);
 
         UniValue sf(UniValue::VARR);
-        // v2 non-ceasable sc
-        if (info.isNonCeasing()) {
-            for(const auto& entry: info.scFees_v2)
-            {
-                UniValue o(UniValue::VOBJ);
-                o.pushKV("forwardTxScFee", ValueFromAmount(entry.forwardTxScFee));
-                o.pushKV("mbtrTxScFee", ValueFromAmount(entry.mbtrTxScFee));
+
+        for(const auto& entry: info.scFees)
+        {
+            UniValue o(UniValue::VOBJ);
+            o.pushKV("forwardTxScFee", ValueFromAmount(entry.forwardTxScFee));
+            o.pushKV("mbtrTxScFee", ValueFromAmount(entry.mbtrTxScFee));
+
+            if (info.fixedParams.version == 2)
```

Comment:
Yes, you are right: we should check also for `withdrawalEpochLength` being 0, as this could create inconsistencies. I will check also the other occurrences of this check over the rest of this PR and the original branch, which this one was derived from.